pkg/services/ngalert/api/api_ruler.go

```diff
@@ -500,6 +503,9 @@ func validateQueries(ctx context.Context, groupChanges *store.GroupDelta, valida
 	}
 	if len(groupChanges.Update) > 0 {
 		for _, upd := range groupChanges.Update {
+			if !shouldValidate(upd) {
+				continue
+			}
```

Comment:
Let's do the opposite: validate queries only if `Data` or `Condition` fields changed.